css-text-4/Overview.bs

```diff
@@ -10720,14 +10720,11 @@ Text Spacing Character Classes</h4>
 
 	<dl>
 		<dt><dfn>ideographs</dfn>
-		<dd>Includes all [=typographic character units=] [[CSS-TEXT-3]] whose base character is listed below:
-			<ul>
-				<li>All characters in the range of U+3041 to U+30FF,
-					except those that belong to Unicode Punctuation [P*] [=general category=].
-				<li>CJK Strokes (U+31C0 to U+31EF).
-				<li>Katakana Phonetic Extensions (U+31F0 to U+31FF).
-				<li>All characters that have the Han [=script property=].
-			</ul>
+		<dd>Includes all [=typographic character units=] [[CSS-TEXT-3]]
+		whose base character:
+		* belongs to Unicode Letters [L*], Mark [M*], Symbols [S*], or Numbers [N*] [=general category=], and
+		* has the Han, Hiragana, or Katakana [=script property=], and
```

Comment:
* [Characters removed by this definition](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%5Cu3041-%5Cu30FF%5D%5B%5Cu3041-%5Cu30FF%5D-%5B%3Agc%3DP%3A%5D-%5B%3Asc%3DHira%3A%5D-%5B%3Asc%3DKana%3A%5D-%5B%3Asc%3DHrkt%3A%5D&g=&i=gc%2Csc%2Cea)
* [Characters added by this definition](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3Asc%3DHira%3A%5D%5B%3Asc%3DKana%3A%5D%5B%3Asc%3DHrkt%3A%5D-%5B%3Aea%3DH%3A%5D-%5B%5Cu3041-%5Cu30FF%5D-%5B%5Cu3041-%5Cu30FF%5D&g=&i=gc%2Csc%2Cea)

From these two lists, I think:
* Adding Kana extensions, Hentaigana, etc. is good.
* U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK should not be removed.
* Should Circled Katakana probably be removed?
* Squared Katakana words, I'm not sure. Ask at JLTF?

Reply:
> U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK should not be removed.

In css-text-4, we have this:
> [script property](http://www.unicode.org/reports/tr24/#Values)
    Defined in Unicode Standard Annex #24 [[UAX24]](https://drafts.csswg.org/css-text-4/#biblio-uax24) and given as the Script property in the [Unicode Character Database](https://www.unicode.org/reports/tr44/) [[UAX44]](https://drafts.csswg.org/css-text-4/#biblio-uax44). (**UAs must include any ScriptExtensions.txt assignments in this mapping**.)

Since U+30FC has the ScriptExtensions property set to Hiragana and Katakana, it is not be removed.

But I think that our defintion of "script property" being not just the Unicode script property but also the Script Extensions property is too easy to miss. Maybe, editorially, we should rename it to "script properties" or "script related properties".

> Should Circled Katakana probably be removed?

No strong opinion, from me. Why do you think they should be removed?

Reply:
I agree with @kojiishi that the character U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK should not be removed. It is cl-10 in JLReq and its behavior for spacing is the same as Hiragana, Katakana and Kanji.

I believe circled Katakana should be treated the same as other circled letters, however it is currently a part of Kanji cl-19. We'll discuss this at JLReq TF.

Squared Katakana words is postfixed abbreviations cl-13 in JLReq. It is a part of symbols in the [new simplified proposal](https://github.com/w3c/jlreq/blob/gh-pages/docs/spacing_property/spacing_property.md). We'll revisit this in JLReq TF also.

Reply:
> the character U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK should not be removed

It is not being removed, so we're in agreement.

Reply:
> Since U+30FC has the ScriptExtensions property set to Hiragana and Katakana, it is not be removed.

The ScriptExtensions property is a different property from the Script property. If you want to use that, it's better to be explicit. [UAX#24 3.2](https://unicode.org/reports/tr24/#Assignment_ScriptX_Values) says it has a list only when the list is "a small, enumerable list," so it's a "be careful when you must rely on it" type of the property.

From implementers' point of view, it's a lot more expensive performance-wise. It's (probably) the only property that return a list. Great if we can avoid that when possible.

For this specific case, if the [Unicode Blocks](https://www.unicode.org/Public/UNIDATA/Blocks.txt) works, that's helpful. Is that ok?

Reply:
@kojiishi That's what we did originally, but when Unicode updated they added letters outside the original blocks. So from the spec point of view, it's better to use the script properties. But from an implementation point of view, you probably want to implement it as unicode blocks.

Reply:
As @xfq reported somewhere, the proposal to the Unicode is in progress. I'd like to see how it goes.

I agree that the maintenance when Unicode updates is a concern for us. I think we're in consensus to try to avoid/minimize it, only slightly different opinions on how.

Allow me to confirm, here we're talking about the script extension property, not the script property, correct? I'm assuming you meant so when you wrote "the script properties" but please let me know if I read you wrong.

First, let me correct my previous comment, it wasn't correct. It was based on the past feedback from Myles and Jonathan. But after talking to UTC experts, they preferred script extension over blocks. This is part because while the script extension property isn't easy for browsers to use in a performant way, it's easier for ICU. If we were going to implement this in ICU, it's possible for ICU to support CSS-defined derived properties, as they did for the line break iterators, but Unicode is more natural place.

Another point. CSS defining simple derived properties, combination of only one or two properties, is fine, but when it goes beyond certain complexity, the risk of overfitting increases, and thus the possibility of maintenance to adopt new Unicode versions will increase, so we may not be achieving our original goal. IIRC this was the primary reason for UAX#50, feedback from both CSS (John) and Unicode (Eric).

Seeing @frivoal's improved proposal and other feedback, I think this properly is likely to exceed the limit. If UTC can accept the proposal, I think it's the best way to go.